tools/c7n_azure/c7n_azure/resources/storage.py

```diff
@@ -465,3 +465,43 @@ def update_logging(storage_type, storage_account, logging_settings, session=None
 
         return getattr(client, 'set_{}_service_properties'
                        .format(storage_type))(logging=logging_settings)
+
+
+@Storage.action_registry.register('require-secure-transfer')
+class RequireSecureTransferAction(AzureBaseAction):
+    """Action that updates the Secure Transfer setting on Storage Accounts.
+    Programatically, this will be seen by updating the EnableHttpsTrafficOnly setting
+
+    :example:
+
+        Set Secure Transfer Required to True
```

Comment:
(minor)I would phrase this in a way that is more scenario based.

"Turns on `Secure transfer required` for all storage accounts."